### PR TITLE
sji animation bug fix

### PR DESCRIPTION
**Pull request by shelbej**
.plot() and .peek() should now return cube animations
memmap can now be turned on/off
added a .percentile() function to maps for statisical analysis, similar to .max(), .mean(), etc

### CODE REVIEW

**Comment by DanRyanIrish** [comment]
Hi @shelbej.  How's this PR coming along?

File: irispy/sji.py [hunk 1]
```diff
@@ -502,16 +515,16 @@ def annotate_frame(i):
             axes.set_title("{s.name}".format(s=self[i]))
 
             # x-axis label
-            if self[0].coordinate_system.x == 'HG':
-                xlabel = 'Longitude [{lon}'.format(lon=self[i].spatial_units.x)
+            if self[0].coordinate_system.axis1 == 'HG':
```
Comment (Cadair) [inline irispy/sji.py]: This should be re-written to use this: https://github.com/sunpy/sunpy/blob/master/sunpy/visualization/visualization.py#L38 helper function from SunPy. As map now uses here: https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L1720

File: irispy/sji.py [hunk 2]
```diff
@@ -539,9 +552,10 @@ def updatefig(i, im, annotate, ani_data, removes):
             # norm.autoscale_None(ani_data[i].data)
             im.set_norm(norm)
 
-            if wcsaxes_compat.is_wcsaxes(axes):
+            if wcsaxes_compat.is_wcsaxes(im.axes):
                 im.axes.reset_wcs(self[i].wcs)
-                wcsaxes_compat.default_wcs_grid(axes)
+                wcsaxes_compat.default_wcs_grid(im.axes, self[i].spatial_units,
+                                             self[i].coordinate_system)
```
Comment (Cadair) [inline irispy/sji.py]: indentation is wrong

File: irispy/sji.py (hunk 1 above)
Comment (shelbej) [reply irispy/sji.py]: fixed

File: irispy/sji.py (hunk 2 above)
Comment (shelbej) [reply irispy/sji.py]: fixed

File: irispy/sji.py [hunk 3]
```diff
@@ -135,20 +135,24 @@ class SJICube(object):
...
 
-    def __init__(self, input):
+    def __init__(self, input, memmap=False):
         """Creates a new instance"""
         if isinstance(input, str):
-            fits = pyfits.open(input, memmap=True, do_not_scale_image_data=True)
-            # TODO find a new masking value for unscaled data
-            #self.data = np.ma.masked_less_equal(fits[0].data, 0)
+            hdulist = fits.open(input, memmap=memmap, do_not_scale_image_data=memmap)
             self.data = fits[0].data
```
Comment (DanRyanIrish) [inline irispy/sji.py]: This line errors.  I think you mean ```self.data = hdulist[0].data```

Comment (shelbej) [reply irispy/sji.py]: fixed

**Pull request closed without merging**